### ISSUE TITLE
Will it wear off my disk drive?

### ISSUE THREAD

**Issue by snimavat** [typicode/lowdb#602]
It seem to write to file entire db on every write operation.
So if i have a db with few 1000 entries, 
and i insert a new record 1001, 
it will write all 1001 records back to file

with frequent data updates, this can wear off the disk, isnt it

**Comment by movy** [comment]
From readme:

```js
// Alternatively you can call db.write() explicitely later
// to write to db.json
db.data.posts.push('hello world')
await db.write()
```

Basically, instead of using `db.update` method, you separate data updates and writes, so you can choose when write-to-disk actually takes place. 

In my production app I update lowdb-data all the time, but I write to disk only once in 5 mins AND whenever my app restarts / crashes. The latter is done via signal handlers:

```js
exitEmitter.on('exit', this.#exitHandler.bind(null, { DB }))
process.on('SIGINT', this.#exitHandler.bind(null, { DB }))
process.on('SIGTERM', this.#exitHandler.bind(null, { DB }))
process.on('beforeExit', async () => await this.#exitHandler.bind(null, { DB }))

async #exitHandler({ DB }) {
    const message = '⚠️ ⚠️ ⚠️  has exited, check logs!!! Saving databases before restart ⚠️ ⚠️ ⚠️'
    await DB.writeAllDatabases()
    await sleep(1500)
    process.exit()
}
```
The app uses and updates ~100 different lowdb objects (heavy multithreading), and to avoid io-related stampede every 5 mins, I slightly randomise writing schedules.
 
I have been using lowdb for over a year and never lost any data on restart. Disk writes are neither performance nor wear-and-tear issue with such architecture.

Basically, for small objects / arrays (up to several Mb of json worth) I love it.

**Comment by wooly99** [comment]
> In my production app I update lowdb-data all the time, but I write to disk only once in 5 mins AND whenever my app restarts / crashes. The latter is done via signal handlers:

Your solution is really great and it has inspired me a lot! I also found a very useful library called [signal-exit](https://github.com/tapjs/signal-exit) which can help write less code to achieve exit writing. I highly recommend it to you.

**Comment by wooly99** [comment]
> > In my production app I update lowdb-data all the time, but I write to disk only once in 5 mins AND whenever my app restarts / crashes. The latter is done via signal handlers:
> > [@movy](https://github.com/movy)
> > Your solution is really great and it has inspired me a lot! I also found a very useful library called [signal-exit](https://github.com/tapjs/signal-exit) which can help write less code to achieve exit writing. I highly recommend it to you.

@movy Sorry, I made a mistake in my recommendation. Your implementation is correct. I found that this library doesn't support executing an asynchronous callback function... Asynchronous file writing doesn't work.
